nibiru/utils.py

```diff
@@ -175,3 +178,73 @@ def toPbTimestamp(dt: datetime):
     ts = Timestamp()
     ts.FromDatetime(dt)
     return ts
+
+
+def merge_list(list_of_dict: Iterable[Dict]) -> Dict:
+    """
+    Merge a list of dictionnary together.
+
+    ```
+    >>> d1 = dict(x=0, y=dict(a=0, b=1), z=dict(x=1))
+    >>> d2 = dict(y=dict(c=2), z=dict(y=1))
+    >>> merge_list(d1, d2)
+
+    {
+        "x": 0,
+        "y": {"a": 0, "b": 1, "c": 2},
+        "z": {
+            "x": 1,
+            "y": 1,
+        },
+    }
+    ```
+
+    Args:
+        list_of_dict (Iterable[Dict]): The list of dictionaries to merge
+
+    Returns:
+        Dict: Dictionaries merged together.
+    """
+
+    return reduce(
+        lambda x, y: MERGER.merge(x, y),
+        list_of_dict,
+    )
+
+
+def transform_java_dict_to_list(events: Dict) -> List[Dict]:
```

Comment:
`message["result"]["data"]["value"]["TxResult"]["result"]["log"]` - here events are stored in a better way. No need to parse java-like stuff.

This is how liquidator is parsing it.

![image](https://user-images.githubusercontent.com/1250307/187917790-8ddc9d1f-4897-4790-8999-1abd8d1a6a8f.png)

Reply:
I simplified the logic using this! thanks